standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/StatObjectConverter.java

```diff
@@ -1064,6 +1118,9 @@ public static void setFieldsIntoOldStats(ColumnStatisticsObj oldStatObj,
       if (newDecimalStatsData.isSetBitVectors()) {
         oldDecimalStatsData.setBitVectors(newDecimalStatsData.getBitVectors());
       }
+      if (newDecimalStatsData.isSetHistogram()) {
+        oldDecimalStatsData.setHistogram(newDecimalStatsData.getHistogram());
+      }
```

Comment:
should also deal with date and timestamp type?

Reply:
Yes, absolutely, thanks for catching that, added.